utils/relevancy_metrics.py

```diff
@@ -0,0 +1,35 @@
+import os
+import subprocess
+import time
+
+
+def get_map_mrr(qids, predictions, labels, device=0):
+    """
+    Get the map and mrr using the trec_eval utility.
+    qids, predictions, labels should have the same length.
+    device is not a required parameter, it is only used to prevent potential naming conflicts when you
+    are calling this concurrently from different threads of execution.
+    :param qids: query ids of predictions and labels
+    :param predictions: iterable of predictions made by the models
+    :param labels: iterable of labels of the dataset
+    :param device: device (GPU index or -1 for CPU) for identification purposes only
+    """
+    qrel_fname = 'trecqa_{}_{}.qrel'.format(time.time(), device)
+    results_fname = 'trecqa_{}_{}.results'.format(time.time(), device)
+    qrel_template = '{qid} 0 {docno} {rel}\n'
+    results_template = '{qid} 0 {docno} 0 {sim} mpcnn\n'
+    with open(qrel_fname, 'w') as f1, open(results_fname, 'w') as f2:
+        docnos = range(len(qids))
+        for qid, docno, predicted, actual in zip(qids, docnos, predictions, labels):
+            f1.write(qrel_template.format(qid=qid, docno=docno, rel=actual))
+            f2.write(results_template.format(qid=qid, docno=docno, sim=predicted))
+
+    trec_out = subprocess.check_output(['../utils/trec_eval-9.0.5/trec_eval', '-m', 'map', '-m', 'recip_rank', qrel_fname, results_fname])
```

Comment:
I will make this path absolute so it works regardless where the cwd is - will address in next commit to this PR.